### ISSUE TITLE
Marketing of Dask talks at events

### ISSUE THREAD

**Issue by mikebeaumont** [dask/community#5]
There are a growing number of events (conferences, meetups, workshops, hackathons) where Dask, RAPIDS, and NVIDIA intersect. As the RAPIDS Product Marketing Manager, who works for NVIDIA, I want to start a discussion on how to market, promote, and collaborate on this work. I believe that the Dask project can benefit from broader exposure without compromising it's duty to the Dask community and open source. 

As an example, @pentschev is speaking at Euro SciPy in September (https://pretalx.com/euroscipy-2019/talk/9DPFGM/). This talk discusses Dask, RAPIDS, CuPy, Numba, and GPUs. I think there's an opportunity for Dask to lead a promotion of Peter's talk on Twitter and tag the others for further amplification.

**Comment by shoyer** [comment]
👍 I would support promoting any talks that highlight dask through the project's social media channels.

**Comment by jrbourbeau** [comment]
I agree with @shoyer about using Dask's social media accounts (which, as far as I know, is just https://twitter.com/dask_dev) to promote talks or blog posts which highlight the project. It's already fairly common for the Dask twitter account to like or retweet other tweets which link to relevant talk materials or blog posts. Here are a few recent instances:

- https://twitter.com/reach_vb/status/1162988693453340672
- https://twitter.com/insideHPC/status/1158754194091700224
- https://twitter.com/stsievert/status/1149407244485844993
- https://twitter.com/HammanHydro/status/1134275348000956416

**Comment by mrocklin** [comment]
Thanks @mikebeaumont for bringing this up.  And thanks @shoyer and @jrbourbeau for responding.

I think that @mikebeaumont 's post can be reduced to two questions:

1.  Is it ok/a good idea to coordinate marketing efforts between affiliated companies like NVIDIA and the Dask marketing resources like the twitter account, blog, and so on.  

    It seems like so far people are generally on board with this.

2.  Operationally how does a marketing professional like @mikebeaumont do this coordination?

    It would be good to hear thoughts on how best to enable this.

One solution to the second part is for Mike to send me things that he wants us to tweet out (we work together) and I tweet them out.  This puts me in a little bit of an odd situation personally and professinoally, because now I'm navigating between my employer and my role in the Dask community.  It also leaves me open to seeming like a company tool, which would be bad both for myself personally, and for the project generally I think.  I would love for us to find a way that didn't involve direct NVidia employees.

I'll also add a third question:

3.   How do we extend this engagement/partnership not just to NVidia , but to other highly engaged companies as well, like Anaconda and QuanSight?

**Comment by mrocklin** [comment]
No one seems to be engaging here.  @mikebeaumont is there anything else that you wanted to bring up here?  Do any Dask maintainers have suggestions on the questions above?

**Comment by TomAugspurger** [comment]
My preference for handling these kinds of situations would be

1. The person promoting the content (say @mikebeaumont in this case) asks the maintainers of the twitter account to publicize something. This can be done in the dask-dev Gitter, or perhaps in a google-group / mailing list that all the owners of the dask-dev account are subscribed to.
2. A person without a conflict of interest decides to actually Tweet (or whatever) the content. And conflict of interest here can be pretty narrow I think (does the person work at the same company as the thing being promoted / requesting the promotion).

I think if we write up some clear guidelines about how the person Tweeting determines if something is appropriate then this should work well enough (for us. Would be curious to hear from @mikebeaumont if this would work from their point of view).

**Comment by jacobtomlinson** [comment]
Here's an over-engineered solution.

How about having a repo that behaves much like [conda forge stages recipes](https://github.com/conda-forge/staged-recipes) repo. Someone PRs a text file containing the contents of a tweet. Content is iterated on using GitHub review features. On merge a bot tweets on our behalf and cleans up the repo.

This is a software engineers solution and appreciate that this may not appeal to everyone. I've had mixed success with using Jekyll in a multi-disciplinary team for the same reason.

**Comment by rgommers** [comment]
@jrbourbeau and myself discussed this with Troy, Quansight's marketing lead. We are supportive of the ideas in this issue about collaborating between Dask and companies engaged in its development, and would love to participate. @TomAugspurger's suggestion on how to handle this in practice seems simple and appropriate.

**Comment by mUtterberg** [comment]
Is the idea behind the repo-esque workflow to maintain transparency? If that’s not a concern, it might be easier to direct tweet requests to @dask_dev DMs.

**Comment by jacobtomlinson** [comment]
@mUtterberg yes that was my intention. Given that Dask maintainers work for different organizations it is important to get unbiased peer review.

However I also agree that this may not be a strong enough concern to warrant a technical solution. @TomAugspurger suggested Gitter which might also be a good place for discussion. Not all maintainers have access to @dask_dev DMs.

**Comment by mUtterberg** [comment]
Is there an avenue that is friendly to non-engineers submitting tweets? I’m not sure what the target group is, but something like survey monkey or google forms could be a way for people to submit tweets for cross-promotion in a non-PR way. I’m not exactly sure how that would fit with the repo/issue thought train, though.

**Comment by TomAugspurger** [comment]
I think most of us (Jim, Matt, myself) on the call were in favor of a low-tech solution. Someone with dask-related content contacts us, and we'll manually send it out.

As for how they contact us, our options are

- GitHub issue
- Gitter channel
- Email to a googlegroup 
- Google Forms

**Comment by mrocklin** [comment]
Marissa, which of those options do you think would make
non-software-developer folks comfortable?  And which would you prefer
personally?

**Comment by mUtterberg** [comment]
Even as a dev, I’m not familiar with gitter (we use GitLab at work, so that’s where I spend most of my time).

IMO, the google form or email to a google group options sound equally accessible to non-devs & devs alike. Might be worth putting out a quick (Twitter?) poll to get wider feedback on that, though.

**Comment by mikebeaumont** [comment]
I'm a fan of a simple google form that guides the community through this process. The more open, simple, accessible, the better in my opinion.

**Comment by TomAugspurger** [comment]
Thanks @mikebeaumont.

On https://twitter.com/adbreind/status/1170757153184829440?s=21 in particular

1. Adam apparently isn't affiliated with Nvidia, so I think any team member could choose to retweet it. That said, it's kinda cross-promoting rapids, so maybe it'd be better from someone outside Nvidia to retweet it
2. It's dask-related, so it's relevant
3. I skimmed through the content to make sure there isn't anything objectionable

So at that point I would be comfortable retweeting it. Thoughts on that process from other dask maintainers?

**Comment by mrocklin** [comment]
Copying the tweet here for convenience:

> Go see @datametrician speak on @rapidsai tomorrow @ApacheCon if you can! 
> Meantime, here’s my post on large-scale ML beyond Hadoop/Spark with @dask_dev and RAPIDS. Do more, easier and faster, with your valuable data! 

This tweet is interesting because it has two separate messages.  In this particular case, I'm in favor of the second line of the tweet, but not so hot on the first line.  I personally probably wouldn't feel comfortable tweeting this because of the association with my employer, but I wouldn't be particularly against anyone else retweeting it.

**Comment by mrocklin** [comment]
I think that one of the reasons that the first line turns me off is that it instructs people to do something.  I would be more comfortable if it said something like "I'm really excited to see ...".

I mention this just because it might be interesting to include in tone suggestions.  Expressing facts, opinions, or states of excitement makes things feel safer to me than telling people to do things.  In this case, obviously Adam doesn't actually expect people to follow his instructions, but the language is slightly closer to that end.

**Comment by pentschev** [comment]
> I think that one of the reasons that the first line turns me off is that it instructs people to do something. I would be more comfortable if it said something like "I'm really excited to see ...".
> 
> I mention this just because it might be interesting to include in tone suggestions. Expressing facts, opinions, or states of excitement makes things feel safer to me than telling people to do things. In this case, obviously Adam doesn't actually expect people to follow his instructions, but the language is slightly closer to that end.

By no means to criticize the way you feel about these things, but I just want to say that this is very subjective, and I think no matter how you write it, some people will be more likely to enjoy one way or the other.

Only to offer an opposite point-of-view (mine): I don't ultimately care if he wrote the way he did, or the way you said you prefer ("I'm really excited to see..."), or a third way, as long as the message is clear, either way I don't think it would have changed the way I felt about the tweet as a whole. However, if I carefully stop to think about both (which I did because it's the subject here), I feel that the original tweet is more natural, in the way that it was just someone writing what came to mind, rather than carefully thinking each word. Anyway, I may not be the average marketing target too. :)

**Comment by mrocklin** [comment]
Sure.  Fair enough.

**Comment by TomAugspurger** [comment]
As a trial: I've made a short google form that looks like

![Screen Shot 2019-09-30 at 9 48 34 AM](https://user-images.githubusercontent.com/1312546/65889829-87250e00-e367-11e9-85ad-ab286de95aa1.png)

That'll be dumped into a google sheet, and the dask maintainers will be notified when the sheet is updated. Does that seem reasonable to people? Any other fields we should add (a "comments field for arbitrary notes from the submitter?")

FYI @mrocklin, the content in the example is https://blog.dask.org/2019/09/30/dask-hyperparam-opt. I wasn't able to see the tweets that you had scheduled.

**Comment by jacobtomlinson** [comment]
Might be nice to have a field for the submitter to suggest the full tweet too? They may want to highlight something specifically from the post or make a comment on it. It would also reduce the effort on the maintainer as they would not have to write the copy themselves.

There may also be multiple items of related content that could be highlighted together. Like [this recent tweet](https://twitter.com/dask_dev/status/1176476877889908737?s=20) for example.

Perhaps a freeform field for the whole tweet which you could just copy-pasta into twitter would be easier?

**Comment by datametrician** [comment]
> As a trial: I've made a short google form that looks like
> 
> ![Screen Shot 2019-09-30 at 9 48 34 AM](https://user-images.githubusercontent.com/1312546/65889829-87250e00-e367-11e9-85ad-ab286de95aa1.png)
> 
> That'll be dumped into a google sheet, and the dask maintainers will be notified when the sheet is updated. Does that seem reasonable to people? Any other fields we should add (a "comments field for arbitrary notes from the submitter?")
> 
> FYI @mrocklin, the content in the example is https://blog.dask.org/2019/09/30/dask-hyperparam-opt. I wasn't able to see the tweets that you had scheduled.

I like this... I will test it out right now.

**Comment by datametrician** [comment]
> I like this... I will test it out right now.
Or I won't... is there a link to this form somewhere?

**Comment by TomAugspurger** [comment]
Sorry @datametrician, I didn't publish the link.

https://docs.google.com/forms/d/e/1FAIpQLSfnc20V6VFSvuU3SN4bym9oS-gbstUbMpg0zboN66ao42KnmQ/viewform?usp=sf_link

**Comment by datametrician** [comment]
After trying out the google form a few times, it would be nice if there was an approved or denied response. I know I’ve been in limbo about a few, “did someone see it”, “was an action taken”, etc... where a simple thumbs up or down would be very useful.

**Comment by TomAugspurger** [comment]
I think I'm the single point of failure on the form right now. Added a few
other maintainers.

I don't think there'd be a way to provide feedback to the submitter via
Google Forms unfortunately.

**Comment by datametrician** [comment]
Adding more maintainers is great! Did you ever get a chance to review my last couple?

**Comment by jacobtomlinson** [comment]
Is this still the process for proposing tweets? Should I fill in the form or have we moved to something else?

**Comment by jacobtomlinson** [comment]
ping @mrocklin

**Comment by mrocklin** [comment]
I know that I personally am not tracking the form.  Tom may be, but I don't think that this went anywhere.

I went to go add you in TweetDeck and it looks like you were already added, and that there is a pending request.  Maybe try logging into tweet deck and clicking on Accounts in the lower left?

**Comment by jacobtomlinson** [comment]
Thanks @mrocklin. I'm more than happy to be able to tweet from the Dask account. 

However in the case where I would want to tweet about RAPIDS or GPUs it would be appropriate to have a non-NVIDIA employee approve it.

**Comment by TomAugspurger** [comment]
I haven't been monitoring it recently. Can we just use this GitHub thread?

**Comment by mrocklin** [comment]
+1 to using this github thread

**Comment by hugobowne** [comment]
We've got an event w/ @TomAugspurger, @mrocklin, and myself this Thursday on _Scalable Machine Learning in Python_.

Is this something that Dask could promote on twitter?

It wouldn't make sense for Matt to tweet it from dask_dev as it's a Coiled event. 

https://twitter.com/hugobowne/status/1292824089531310088

https://www.eventbrite.com/e/scalable-machine-learning-in-python-science-thursday-tickets-116369928589

**Comment by martindurant** [comment]
Personally, I think it's OK to promote dask core devs in conversation about dask, even when hosted by a commercial company. The case of Coiled is a little special, so I can see why you would want to be careful - dask itself doesn't want to be seen as promoting or pushing people towards Coiled, but I think we can have wording clear enough.

**Comment by mrocklin** [comment]
Yeah, I think that it's ok too.  What I think probably isn't ok is me making a unilateral decision to promote for-profit activity that benefits myself.  

If people here want to say something like "The Coiled Science Thursday events seem to be community-based enough and in-line enough with Dask's marketing tone that we're comfortable with a blanket 'go ahead' policy in the future until rescinded" then I would go ahead and retweet by default, or add some community-based messaging around the events from the `@dask_dev` twitter handle.  I think that I probably need blessing from other folks before doing that though.

**Comment by martindurant** [comment]
> The Coiled Science Thursday events seem to be community-based enough and in-line enough with Dask's marketing tone that we're comfortable with a blanket 'go ahead' policy in the future until rescinded

I believe yes

**Comment by jacobtomlinson** [comment]
Seconded

**Comment by jsignell** [comment]
Yeah, I agree with Martin that conversations/work by reliable people about dask are worthy of a retweet.

**Comment by datametrician** [comment]
I support this, but my support may be seen as self-serving as RAPIDS uses Dask immensely and NVIDIA loves to promote RAPIDS.

**Comment by shoyer** [comment]
+1 for advertising this Coiled event on Twitter.

**Comment by hugobowne** [comment]
We have another Science Thursday this week with @jacobtomlinson 
https://www.eventbrite.com/e/deploying-and-scaling-data-science-tools-on-distributed-systems-tickets-117357554605
From the above, it sounds as though you are all ok with dask_dev promoting these events. I'll ping Matt to do so.
Feel free to comment with any concerns // questions.

**Comment by TomAugspurger** [comment]
I took care of it.

**Comment by jacobtomlinson** [comment]
Thanks @TomAugspurger

**Comment by hugobowne** [comment]
I'm super excited about Science Thursday this week and it seems in line with the above so if @dask_dev could share, that would be awesome.

https://www.eventbrite.com/e/data-processing-at-blue-yonder-one-supply-chain-at-a-time-tickets-127808266967

Supply Chain Analytics! And "How to build a ML data pipeline at terabyte scale using parquet, dask and kartotek". Wow!

**Comment by hugobowne** [comment]
Happy New Year, all!

We have a Science Thursday with Eric Dill this week, a comparative analysis of Dask, Spark, and RAPIDS. If this seems in line with the above, a twitter share would be awesome: https://twitter.com/CoiledHQ/status/1344312292415066112

Thanks!

**Comment by jakirkham** [comment]
> Happy New Year, all!
> 
> We have a Science Thursday with Eric Dill this week, a comparative analysis of Dask, Spark, and RAPIDS. If this seems in line with the above, a twitter share would be awesome: https://twitter.com/CoiledHQ/status/1344312292415066112
> 
> Thanks!

Happy New Year!

Sounds good to me. Though perhaps I'm biased. Maybe @TomAugspurger or @martindurant have thoughts? 🙂

**Comment by martindurant** [comment]
I have no objection

**Comment by jakirkham** [comment]
We are presenting on an image processing library to use with Dask ( https://twitter.com/quasiben/status/1380590542891986944 ). Curious if that would be ok to retweet 🙂

cc @quasiben

**Comment by martindurant** [comment]
The tweet dosn't mention Dask, but I don't see why we can't retweet it. If I don't hear otherwise before the end-of-day, I can do it.

**Comment by mrocklin** [comment]
+1

**Comment by martindurant** [comment]
done

**Comment by jakirkham** [comment]
Thanks Martin! 😄

**Comment by jakirkham** [comment]
We did a SciPy talk recently on cuCIM and included some examples using with Dask & Napari. Tweeted out the YouTube video here ( https://twitter.com/NVIDIAAI/status/1419795697272909826 ). Would it be ok to retweet from the Dask account?

**Comment by jakirkham** [comment]
^ @martindurant?

**Comment by martindurant** [comment]
Thanks for the ping. I'll make it happen - and then I'll watch the video! Still want deeply to get napari and Intake connecting, talking the same language...

**Comment by jakirkham** [comment]
Thanks Martin! 😄

Agreed that would be great.
